app/controllers/trainees/check_details_controller.rb

```diff
@@ -4,9 +4,21 @@ module Trainees
   class CheckDetailsController < ApplicationController
     include Breadcrumbable
 
+    before_action :ensure_trainee_is_draft!
+
     def show
-      @trainee = Trainee.from_param(params[:id])
-      save_origin_page_for(@trainee)
+      authorize trainee
+      save_origin_page_for(trainee)
+    end
+
+  private
+
+    def trainee
+      @trainee ||= Trainee.from_param(params[:id])
+    end
+
+    def ensure_trainee_is_draft!
```

Comment:
Should we move these methods into a module to re-use? We've defined this twice and they look identical?

Reply:
@darokel - good point! I've moved both `ensure_trainee_is_draft!` and `ensure_trainee_is_not_draft!` into the `ApplicationController` so that they can be dropped into controllers as before actions.